### PR TITLE
Joy drops root privileges to joy user in online packet-capture mode

### PR DESCRIPTION
**Pull request by brilong**
Added configuration option username (joy is default).  When joy starts in online mode, it opens the network interface as root and immediately drops to joy user for writing JSON and log files.
Package script has also been improved to minimize requirements based on compression option; if gzip compression is used, only require zlib.  If bzip2 compression is used, only require bzip2.
Also updated the default config file to add documentation on various configuration items (ssh, useranon, username, and show_config).

### CODE REVIEW

File: src/joy.c [hunk 1]
```diff
@@ -845,6 +856,39 @@ int main (int argc, char **argv) {
...
 
+#ifndef _WIN32
+        /*
+         * drop privileges once pcap handle exists
+         */
+        pw = getpwnam(config.username);
+        if (pw) {
+            if (initgroups(pw->pw_name, pw->pw_gid) != 0 ||
+                setgid(pw->pw_gid) != 0 || setuid(pw->pw_uid) != 0) {
+                fprintf(stderr, "error: could not change to '%.32s' uid=%lu gid=%lu: %s\n",
```
Comment (fliphil) [inline src/joy.c]: How does the .32 change the format of string here? Can you get away just %s?

**Comment by fliphil** [comment]
Can anyone change to the new "joy" user, i.e. read and write the directories owned by it? If I were to provide my username instead, would it require my credentials to access?

File: src/joy.c (hunk 1 above)
Comment (brilong) [reply src/joy.c]: .32 specifies the precision and I borrowed this from the tcpdump code that handles dropping root privileges.

**Comment by brilong** [comment]
Only users with sudo privileges or root would be able to change to the joy user and read/write the data and log directories.  The data and log directories default permissions are 775 owned by user and group joy.  If another user wanted to write into those directories, he would need to be in the joy group.  This brings up an interesting point about our install script used outside the binary packages.  I need to ensure proper permissions inside the install script for people that manually install joy using install-sh instead of binary packages.

**Comment by brilong** [comment]
I have uploaded various changes to the install and uninstall scripts and verified the MacOS X and Linux Joy drops privileges properly and logs all data and logs as the joy user.  I also changed the options.cfg exe option to 0 since I had problems with it on Mac.

**Pull request merged**